scripts/hermes-binary.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+[ -z "$HERMES_PATH" ] && export HERMES_PATH="$REACT_NATIVE_DIR/../hermes-engine-darwin"
```

Comment:
By default, Hermes engine will get hoisted and will live next to React Native root. 

Also, decided to ask users to set path to a folder, not to a Hermes executable. I believe it is easier to speak to React Native developers in terms of `npm` packages, rather than asking them to set a path to an executable (they need to know that there's "destroot/bin" folder in case things go wrong). 

Relying on a package makes it a bit easier.

Reply:
Theoretically, we could check if HERMES_PATH="$REACT_NATIVE_DIR/node_modules/hermes-engine-darwin" exists, which is true for RNTester or when hoisting is disabled (and Hermes dependency lives inside React Native own node_modules).

Shall we? Or the error message below is good enough?

Reply:
Sure, seems a fair assumption 👍

Reply:
I’m a little unclear on the need for this to be a separate script, can you elaborate on that?

Reply:
Just followed a similar pattern that we currently have for `node-binary.sh`. Not sure if that's the ideal way going forward, but I felt like this was the pattern.

Reply:
Well, I guess the same question applies, why was that necessary? (Perhaps a git blame will give us the reason.) I personally don't like adding indirection when it isn’t serving a reason, but also don’t feel strong about this; just want to understand 😄

Reply:
Looks like it was introduced here, but I couldn’t find a reason for making it a separate file https://github.com/facebook/react-native/pull/24156

Reply:
Let's move to a single file.